framework_tvb/tvb/core/services/import_service.py

```diff
@@ -318,30 +268,110 @@ def import_project_operations(self, project, import_path, dt_burst_mappings=None
+                    if old_operation_folder != new_operation_path:
+                        # Delete folder of the new operation, otherwise move will fail
+                        shutil.rmtree(new_operation_path)
+                        shutil.move(old_operation_folder, new_operation_path)
+
+                    operation_datatypes = self._load_datatypes_from_operation_folder(new_operation_path,
+                                                                                     operation_entity,
+                                                                                     datatype_group)
+                    self._store_imported_datatypes_in_db(project, operation_datatypes, dt_burst_mappings,
+                                                         burst_ids_mapping)
+                    imported_operations.append(operation_entity)
+            else:
+                start_date = datetime.now()
+                alg = VIEW_MODEL2ADAPTER[type(view_model)]
+
+                # import operation only if there is a algolithm
+                if alg:
+                    op = self.get_new_operation_for_view_model(project, view_model, alg.id)
+                    op.meta_data = '{"from": "Import"}'
+                    op.status = STATUS_FINISHED
+                    op.create_date = start_date
+                    op.start_date = start_date
+                    op.algorithm = alg
+                    op.visible = True
+                    op.completion_date = datetime.now()
+                    operation_entity = dao.store_entity(op)
+                    imported_operations.append(operation_entity)
+
+                    # Store the DataTypes in db
+                    if dt_paths:
+                        self._store_datatypes_from_path_in_db(dt_paths, project.id, op.id)
 
-            # Rename operation folder with the ID of the stored operation
-            new_operation_path = FilesHelper().get_operation_folder(project.name, operation_entity.id)
-            if old_operation_folder != new_operation_path:
-                # Delete folder of the new operation, otherwise move will fail
-                shutil.rmtree(new_operation_path)
-                shutil.move(old_operation_folder, new_operation_path)
+        return imported_operations
 
-            operation_datatypes = self._load_datatypes_from_operation_folder(new_operation_path, operation_entity,
-                                                                             datatype_group)
-            self._store_imported_datatypes_in_db(project, operation_datatypes, dt_burst_mappings, burst_ids_mapping)
-            imported_operations.append(operation_entity)
+    @staticmethod
+    def _get_operation_paths(paths):
+        path_list = [f.path for f in os.scandir(paths) if f.is_dir()]
+        sorted_dir = {}
+        for path in path_list:
+            op_number = os.path.basename(path)
+            sorted_dir[int(op_number)] = path
```

Comment:
This convert to `int` produces problems when importing links. We should order ops differently. Maybe by `ViewModel.create_date` field (similar with the previous process

Reply:
I removed this part, but the code now has a big TODO in line 229, as the ordering of the operations is missing.
Instead I added field `create_data` in every ViewModelH5 file. We should read that and put it on the Operation.create_data before we save it. Then we could order Operations by the create_date (either read from `Operation.xml` or from `ViewModelH5`) This should give a correct order, I think.